docs/design-proposals/node/Unified-Vertical-Scaling-for-Container-and-VM-Pods

```diff
@@ -0,0 +1,321 @@
+---
+title: In-place Update of Pod Resources for Container and VM Pods
+authors:
+  - "@vinaykul"
+  - "@yb01"
+---
+
+# In-place Update of Pod Resources
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [API Changes](#api-changes)
+    - [Resize Policy](#resize-policy)
+    - [CRI Changes](#cri-changes)
+  - [Kubelet and API Server Interaction](#kubelet-and-api-server-interaction)
```

Comment:
Will we address VM-specific parts in another design doc?  For example, for VM vCPU changes, what's the observed vCPU number changes from guest OS perspective; is there a max limit on dynamically added vCPUs; Do we expect VM memory decreasing work, etc.

Reply:
Checked with Yunwen. He will add it later after this initial version gets merged.